src/commands/doctor.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import { exec } from 'child_process';
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, Lifecycle } from '@salesforce/core';
+import { Doctor as SFDoctor, SfDoctorDiagnosis } from '../doctor';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-info', 'doctor');
+
+export default class Doctor extends SfdxCommand {
+  public static description = messages.getMessage('commandDescription');
+
+  public static examples = messages.getMessage('examples').split(os.EOL);
+
+  // Hide for now
+  public static hidden = true;
+
+  protected static flagsConfig = {
+    command: flags.string({
+      char: 'c',
+      description: messages.getMessage('flags.command'),
+    }),
+    newissue: flags.boolean({
+      char: 'i',
+      description: messages.getMessage('flags.newissue'),
+    }),
+    plugin: flags.string({
+      char: 'p',
+      description: messages.getMessage('flags.plugin'),
+    }),
+  };
+
+  public async run(): Promise<SfDoctorDiagnosis> {
+    let promises: Array<Promise<void>> = [];
+    const doctor = SFDoctor.getInstance();
+    const lifecycle = Lifecycle.getInstance();
+
+    const plugin = this.flags.plugin as string;
+    const command = this.flags.command as string;
+    const newissue = this.flags.newissue as boolean;
+
+    if (command) {
+      const cmdString = this.parseCommand(command);
+      this.ux.log(`Running Command: ${cmdString}\n`);
```

Comment:
I only have the most basic output for now.  I think we can do some cool things with the output to give the user a much better sense of what the doctor is doing.